apps/keyboard/js/layouts/uk.js

```diff
@@ -0,0 +1,85 @@
+Keyboards.uk = {
+  label: 'Ukrainian',
+  shortLabel: 'Uk',
+  menuLabel: 'українська',
+  imEngine: 'latin',
+  types: ['text', 'url', 'email', 'password'],
+  basicLayoutKey: 'АБВ',
+  lang: 'uk',
+  alt: {
+    а: '@',
+    г: 'ґ'
+  },
+  width: 12,
+  keys: [
+    [
+      { value: 'й' }, { value: 'ц' }, { value: 'у' }, { value: 'к' },
+      { value: 'е' }, { value: 'н' }, { value: 'г' }, { value: 'ш' },
+      { value: 'щ' }, { value: 'з' }, { value: 'х' }, { value: 'ї' }
+    ], [
+      { value: 'ф' }, { value: 'і' }, { value: 'в' }, { value: 'а' },
+      { value: 'п' }, { value: 'р' }, { value: 'о' }, { value: 'л' },
+      { value: 'д' }, { value: 'ж' }, { value: 'є' }
+    ], [
+      { value: '⇪', ratio: 1.5, keyCode: KeyEvent.DOM_VK_CAPS_LOCK },
+      { value: 'я' }, { value: 'ч' }, { value: 'с' }, { value: 'м' },
+      { value: 'и' }, { value: 'т' }, { value: 'ь' }, { value: 'б' },
+      { value: 'ю' }, { value: '⌫', ratio: 1.5, keyCode: KeyEvent.DOM_VK_BACK_SPACE }
+    ], [
+      { value: '&nbsp', ratio: 10, keyCode: KeyboardEvent.DOM_VK_SPACE },
+      { value: '↵', ratio: 2, keyCode: KeyEvent.DOM_VK_RETURN }
+    ]
+  ],
+  pages: [undefined, {
+    alt: {
+      '0': ['º'],
+      '1': ['1-ий','1-а','1-е','1-і'],
```

Comment:
I found them here http://intranet.tdmu.edu.ua/data/kafedra/internal/in_mow/lectures_stud/en/nurse/bsn/ptn/4/Ukrainian%20language/%E2%84%96%2009.%20Numerals.%20Dates.%20Ordinal%20numbers.htm , but I'm not Ukrainian and I don't know if it's correct or not